protocol/v2/ssv/validator/committee_queue.go

```diff
@@ -30,24 +30,28 @@ func (c *Committee) HandleMessage(logger *zap.Logger, msg *queue.SSVMessage) {
 		return
 	}
 
-	c.mtx.RLock() // read v.Queues
-	q, ok := c.Queues[slot]
-	c.mtx.RUnlock()
-	if !ok {
-		q = queueContainer{
-			Q: queue.WithMetrics(queue.New(1000), nil), // TODO alan: get queue opts from options
-			queueState: &queue.State{
-				HasRunningInstance: false,
-				Height:             specqbft.Height(slot),
-				Slot:               slot,
-				//Quorum:             options.SSVShare.Share,// TODO
-			},
-		}
+	prepareQueue := func() queueContainer {
 		c.mtx.Lock()
-		c.Queues[slot] = q
-		c.mtx.Unlock()
-		logger.Debug("missing queue for slot created", fields.Slot(slot))
+		defer c.mtx.Unlock()
+
+		q, ok := c.Queues[slot]
+		if !ok {
+			q = queueContainer{
+				Q: queue.WithMetrics(queue.New(1000), nil), // TODO alan: get queue opts from options
+				queueState: &queue.State{
+					HasRunningInstance: false,
+					Height:             specqbft.Height(slot),
+					Slot:               slot,
+					//Quorum:             options.SSVShare.Share,// TODO
+				},
+			}
+			c.Queues[slot] = q
+			logger.Debug("missing queue for slot created", fields.Slot(slot))
+		}
+
+		return q
 	}
+	q := prepareQueue()
```

Comment:
Here we potentially can miss a message if `HandleMessage` is called concurrently 2 or more times(if we just lock `c.mtx` for read and then potentially add new Queue - concurrent caller might do the same), 

not sure if this is a real problem in practice, just seems to me it's better not to make additional assumptions and do this lazy initialization in safe manner (which is what `prepareQueue` does).